ext/reflection/php_reflection.c

```diff
@@ -6190,6 +6387,204 @@ ZEND_METHOD(ReflectionReference, getId)
 }
 /* }}} */
 
+ZEND_METHOD(ReflectionAttribute, __construct)
+{
+}
+
+ZEND_METHOD(ReflectionAttribute, __clone)
+{
+}
+
+/* {{{ proto public string ReflectionAttribute::getName()
+ *	   Returns the name of the attribute */
+ZEND_METHOD(ReflectionAttribute, getName)
+{
+	reflection_object *intern;
+	attribute_reference *attr;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		RETURN_THROWS();
+	}
+	GET_REFLECTION_OBJECT_PTR(attr);
+
+	RETURN_STR_COPY(attr->data->name);
+}
+/* }}} */
+
+static zend_always_inline int import_attribute_value(zval *ret, zval *val, zend_class_entry *scope) /* {{{ */
+{
+	if (Z_TYPE_P(val) == IS_CONSTANT_AST) {
+		*ret = *val;
```

Comment:
```suggestion
		ZVAL_COPY_VALUE(ret, val);
```

But actually ... shouldn't this be?
```suggestion
		ZVAL_COPY(ret, val);
```
Otherwise this will double free the AST.

Reply:
This leads to memory leaks:

```
024+ [Mon May 25 23:52:27 2020]  Script:  '/php-src/Zend/tests/attributes/003_ast_nodes.php'
025+ /php-src/Zend/zend_string.h(141) :  Freeing 0x00007f8788a5aa20 (56 bytes), script=/php-src/Zend/tests/attributes/003_ast_nodes.php
026+ [Mon May 25 23:52:27 2020]  Script:  '/php-src/Zend/tests/attributes/003_ast_nodes.php'
027+ /php-src/Zend/zend_ast.c(815) :  Freeing 0x00007f8788a93900 (32 bytes), script=/php-src/Zend/tests/attributes/003_ast_nodes.php
028+ === Total 2 memory leaks detected ===
````

Reply:
This also needed a zval_ptr_dtor in the failure branch, I've pushed a commit.